### PR TITLE
Refactor Cypress Auth Test to POM and Improve Test Stability 

### LINKED ISSUE #9015: Refactor the Existing Cypress Auth Test
**Description**

Refactor all the existing auth test in the platform to POM approach and use the reusable components

**Task - Part 1**

**Rename and Merge Files:**

- Rename the [auth_spec](cypress/e2e/auth_spec) to homepage_spec
- Rename the [auth.cy.ts](https://github.com/ohcnetwork/care_fe/blob/develop/cypress/e2e/auth_spec/auth.cy.ts) to UserLogin.cy.ts
- Combine [auth.cy.ts](https://github.com/ohcnetwork/care_fe/blob/develop/cypress/e2e/auth_spec/auth.cy.ts) and [forgotpassword.cy.ts](cypress/e2e/auth_spec/ForgotPassword.cy.ts) into UserLogin.cy.ts under separate describe blocks. This keeps tests organized and logically grouped.

**Refactor to POM**

- Import and reuse login credentials from the LoginPage.ts file. If the credentials are not present, add new ones to the LoginPage file as well, and replace any direct usage of cy.loginbyapi with calls to LoginPage.
- Add API verification with cy.intercept before clicking #user-profile-name to ensure the page is fully loaded.
- Convert lines 9 and 10 to POM and create functions for them in the UserProfilePage.ts file.
- Similarly, convert all code from lines 15 to 21 to the POM approach, adding cy.intercept for verification and moving credentials to LoginPage if they are missing.
- Use the reusable helper functions in the [commands.ts](cypress/support/commands.ts) file for the submit button , notification verification, etc... common functions
- Remove the hardcoded wait, and if necessary, replace it with cy.intercept() API verification to avoid flaky tests.
 
Image

**Task - 2**

Image

- Combine all the tests into one single test
- Test Follow: Click on forgot password -> Send password reset link -> Go back to the login page
- Follow all the similar approaches for writing code mentioned in the test - 1

### PR DESCRIPTION
**Pull request by JavidSumra**
## Proposed Changes

- Fixes #9015
- Renamed `auth_spec` to `homepage_spec` and `auth.cy.ts` to `UserLogin.cy.ts`.
- Merged `auth.cy.ts` and `forgotpassword.cy.ts` into `UserLogin.cy.ts`, organizing them under separate `describe` blocks.
- Refactored test to use the Page Object Model (POM) approach, utilizing `LoginPage.ts` for login credentials.
- Added API verification with `cy.intercept` before accessing `#user-profile-name` to ensure page load.
- Converted hardcoded elements and actions to reusable functions in `UserProfilePage.ts` and `commands.ts`.
- Removed hardcoded waits and replaced them with `cy.intercept` for stability.

## Merge Checklist

- [ ] Add specs that demonstrate bug fixes and verify new POM structure.
- [ ] Update [product documentation](https://docs.ohc.network) as needed.
- [x] Ensure all UI text updates are included in I18n files.
- [ ] Prepare and attach a screenshot or demo video for changelog entry.
- [ ] Request peer reviews from @ohcnetwork/care-fe-code-reviewers.
- [ ] Complete QA and confirm test stability.

## Summary by CodeRabbit

- **New Features**
	- Introduced automated tests for user login and password recovery functionalities.
	- Enhanced login functionality with new methods for user interaction and request verification.

- **Bug Fixes**
	- Removed outdated test files for "Forgot Password" and authorization functionalities.

- **Chores**
	- Improved accessibility by adding identifiable attributes to buttons in the login and password recovery forms.

### CODE REVIEW

**Comment by JavidSumra** [comment]
Hey @nihal467 can you please review this PR

File: cypress/e2e/homepage_spec/UserLogin.cy.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+  it("Try login as admin with incorrect password", () => {
+    loginPage.interceptLoginReq();
+    loginPage.loginManuallyAsDistrictAdmin(false);
+    loginPage.verifyLoginReq();
+    cy.verifyNotification("No active account found with the given credentials");
+  });
+});
+
+describe("Forgot Password", () => {
```
Comment (nihal467) [inline cypress/e2e/homepage_spec/UserLogin.cy.ts]: @JavidSumra read the issue carefully, it was mentioned to combine all the three test under forgot password to one single test

**Review by nihal467 — APPROVED** [review]

**Review by khavinshankar — APPROVED** [review]

**Pull request merged**